package interconn;

import "ipc.proto";

enum ConnectionStatus
{
	CONN_OPEN	= 1;
	CONN_CLOSE	= 2;
	CONN_FAILED	= 3;
	CONN_EXIST	= 4;
	CONN_INIT	= 5;
}

message InitInternalConnection
{
	required ipc.IPCName target = 1;
	required string ip = 2;
	required int32 port = 3;
}

message InternalConnectionStatus
{
	required ipc.IPCName target = 1;
	required ConnectionStatus status = 2;
	optional int32 port = 3;
}

message InternalConnectionData
{
	required string id = 1;
	optional bytes data = 2;
}

option optimize_for = LITE_RUNTIME;